### PR TITLE
Merge DsAPI to SapiNT migration

### PR DESCRIPTION
**Pull request by ijdickinson**
This PR is primarily to merge the changes made to this gem to support SapiNT as a back-end API, rather than the original DsAPI. The changes in this PR are already active in the re-built dev service.

In addiition, I have updated the README to reflect the history of the gem.

- Added extra file to main gem file
- SAPINT Converter class added
- Changed default api url
- Dataset endpoint modified
- Query method now converts DSAPI to SAPINT
- Removed explain functionality
- Refactored the converter class
- Moved the json converting part out of the query converter class
- Added the query as Faraday params instead of already in URL
- Explain functionality added
- Using map and reduce instead of each
- Refactored response conversion to a separate class
- Refactored converter and method
- Drying up comparison methods in the converter
- Constructing the SAPINT params as JSON now
- New service methods refactoring
- Fixed old failing tests
- Added dsapi response converter tests
- Fixed nil value bug for prefix removal
- Added sapint converter tests
- Removed api default url
- Deleted unnecessary tests
- Added extra test and small refactoring
- Fixed failing tests due to default url removal
- Remove unused methods
- Returning explain JSON output straight from the API
- Bumped gem version
- Added new changelog entry
- Removing prefix from sort properties
- Returning as an array instead of a hash
- Extracting the dataset name before converting the response
- Figure out the json mode from the query
- Added a flag for json compact mode to the response converter
- Returning different responses based on the json mode
- Added response converter for json compact mode
- JSON complete mode response converter added
- Added dataset name as config param
- Using query terms Hash instead of parsing JSON string
- Changed default port for testing the API connection
- Mimic the QueryGenerator class for the query object
- Updated VCR cassette
- Separated PPD response converter tests
- UKHPI response converter tests added
- App version bump
- Added changelog entry
- Fix readme markdown errors
- Added instructions to run the tests to the readme
- Keyword argument to positional argument
- Get API url for tests from env var
- Updated instructions for running the tests
- Small refactor
- Fixed wrong namespace for rubocop rule
- Update the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+```
 
 And then execute:
 
-    $ bundle
+```sh
+bundle
+```
+
+Also, an API URL needs to be provided for the Service class in order for the gem
```
Comment (joescottdave) [inline README.md:55]: Is this provided as an environment variable? How do you do that?

File: lib/data_services_api/version.rb [hunk 2]
```diff
@@ -2,6 +2,6 @@
 
 # :nodoc:
 module DataServicesApi
-  VERSION = '0.4.5'
+  VERSION = '1.1.0'
```
Comment (joescottdave) [inline lib/data_services_api/version.rb:5]: Odd jump

**Review by joescottdave — APPROVED** [review]
Lgtm

File: lib/data_services_api/version.rb (hunk 2 above)
Comment (ijdickinson) [reply lib/data_services_api/version.rb:5]: Yes, a bit unfortunate. This is a long-running branch that was set up to help the new LR infra work, and had multiple updates to the API-wrapper gem. We bumped to 1.0.0 on the initial update, and then subsequent new features added. Merging back to master probably should have happened a while ago.

File: README.md (hunk 1 above)
Comment (ijdickinson) [reply README.md:55]: Thanks, I've fixed the description to make it clearer

**Pull request merged**